test/test_normal_exceptions.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+
+
+def test_simple(tracer, check_golden_file):
+    tracer.start()
+
+    class CustomException(Exception):
+        pass
+
+    a = ImportError("A")
+    b = CustomException("B")
+    c = AssertionError()
+    print(a)
+    print(b)
+    print(c)
+    c = b
+    print(c)
+    del a
+    del b
+    del c
+    tracer.stop()
+
+
+def test_throw(trace, check_golden_file):
+    @trace
+    def test_throw_inner():
+        with pytest.raises(ZeroDivisionError):
+            a = ZeroDivisionError("A")
+            print(a)
+            raise a
+
+    test_throw_inner()
+
+
+def test_throw_catch(tracer, check_golden_file):
+    tracer.start()
+    b = TypeError("B")
+    try:
+        try:
+            a = RuntimeError("A")
+            raise a
+        except b.__class__ as e:
```

Comment:
it's not entirely clear to me what this part is for, mind elaborate?

Reply:
It's using a merge_stack_items on the error object from raise_args and a stack item containing B, the second except is unnecessary since it's similar to the outer except.

Reply:
Consider breaking this test down to smaller ones, ideally with one-except, and please add proper comments. In any case, I'd want to avoid tests from which readers cannot see the intent at first glance.